pkgs/applications/misc/octoprint/default.nix

```diff
@@ -33,70 +41,37 @@ let
     ]);
   };
 
-  ignoreVersionConstraints = [
-    "Click"
-    "Flask-Assets"
-    "Flask-Babel"
-    "Flask-Principal"
-    "emoji"
-    "flask"
-    "future"
-    "futures"
-    "monotonic"
-    "markdown"
-    "pkginfo"
-    "psutil"
-    "pyserial"
-    "requests"
-    "rsa"
-    "sarge"
-    "scandir"
-    "semantic_version"
-    "watchdog"
-    "websocket-client"
-    "wrapt"
-    "sentry-sdk"
-    "werkzeug" # 0.16 just deprecates some stuff
-  ];
-
-in py.pkgs.buildPythonApplication rec {
+in
+py.pkgs.buildPythonApplication rec {
   pname = "OctoPrint";
-  version = "1.3.12";
+  version = "1.4.0";
 
   src = fetchFromGitHub {
     owner  = "foosel";
     repo   = "OctoPrint";
     rev    = version;
-    sha256 = "1lmqssgwjyhknjf3x58g7cr0fqz7fs5a3rl07r69wfpch63ranyd";
+    sha256 = "1zla1ayr62lkvkr828dh3y287rzj3rv1hpij9kws44ynn4i582ga";
   };
 
   propagatedBuildInputs = with py.pkgs; [
-    awesome-slugify flask_assets rsa requests pkginfo watchdog
-    semantic-version flask_principal werkzeug flaskbabel tornado
-    psutil pyserial flask_login netaddr markdown sockjs-tornado
+    awesome-slugify flask flask_assets rsa requests pkginfo watchdog
+    semantic-version werkzeug flaskbabel tornado
+    psutil pyserial flask_login netaddr markdown
     pylru pyyaml sarge feedparser netifaces click websocket_client
-    scandir chainmap future futures wrapt monotonic emoji
-    frozendict cachelib sentry-sdk typing filetype
+    scandir chainmap future wrapt monotonic emoji jinja2
+    frozendict cachelib sentry-sdk filetype markupsafe
   ] ++ lib.optionals stdenv.isDarwin [ py.pkgs.appdirs ];
 
   checkInputs = with py.pkgs; [ nose mock ddt ];
 
-  postPatch = ''
-    sed -r -i \
-      ${lib.concatStringsSep "\n" (map (e:
-        ''-e 's@${e}[<>=]+.*@${e}",@g' \''
-      ) ignoreVersionConstraints)}
-      setup.py
-  '';
-
   checkPhase = ''
-    HOME=$(mktemp -d) nosetests ${lib.optionalString stdenv.isDarwin "--exclude=test_set_external_modification"}
+    HOME=$(mktemp -d) nosetests --exclude-test=util.test_pip.test_check_setup ${lib.optionalString stdenv.isDarwin "--exclude=test_set_external_modification"}
```

Comment:
`--exclude-tests` does not exist.

Reply:
I believe it does: https://pypi.org/project/nose-exclude/

Reply:
But if that's not part of your checkInputs, how is it pulled in?

Reply:
Also `--ignore-files=test_pip.py` is probably required.

Reply:
And the sentry-sdk tests fail for me, so i disabled them locally. This makes the package build.

Reply:
I'm not sure, but IIRC the exclude flag was properly doing the job at one point while I was tinkering. 

Do you mean you got Octoprint to build entirely?

Reply:
Yes, I did.

Reply:
Just by adding `doCheck = false;` to sentry-sdk? Because that doesn't help for me:

```
Found duplicated packages in closure for dependency 'MarkupSafe':
  MarkupSafe 1.1.1 (/nix/store/qvpsgs0sl3sh3knm5b1ya49x5zlf9v6j-python3-3.7.6/lib/python3.7/site-packages)
  MarkupSafe 1.1.1 (/nix/store/ig1lgbvd518i5yci62mm8hxpsjm9hlvy-python3.7-MarkupSafe-1.1.1/lib/python3.7/site-packages)
 ```
And now this is happening too:
```
Found duplicated packages in closure for dependency 'urllib3':
  urllib3 1.25.8 (/nix/store/qvpsgs0sl3sh3knm5b1ya49x5zlf9v6j-python3-3.7.6/lib/python3.7/site-packages)
  urllib3 1.25.8 (/nix/store/ka86vbis93vyyj3rn494n5yc3ymk9x24-python3.7-urllib3-1.25.8/lib/python3.7/site-packages)
Found duplicated packages in closure for dependency 'six':
  six 1.14.0 (/nix/store/qvpsgs0sl3sh3knm5b1ya49x5zlf9v6j-python3-3.7.6/lib/python3.7/site-packages)
  six 1.14.0 (/nix/store/f2p4y8awg8px77wj2p8x5v8cv4sr7rgw-python3.7-six-1.14.0/lib/python3.7/site-packages)
Found duplicated packages in closure for dependency 'pyparsing':
  pyparsing 2.4.6 (/nix/store/qvpsgs0sl3sh3knm5b1ya49x5zlf9v6j-python3-3.7.6/lib/python3.7/site-packages)
  pyparsing 2.4.6 (/nix/store/kdbxn8yk7v8j36slygdhz59xy1j7s6ii-python3.7-pyparsing-2.4.6/lib/python3.7/site-packages)
Found duplicated packages in closure for dependency 'pyasn1':
  pyasn1 0.4.8 (/nix/store/qvpsgs0sl3sh3knm5b1ya49x5zlf9v6j-python3-3.7.6/lib/python3.7/site-packages)
  pyasn1 0.4.8 (/nix/store/d14dfx619lgcaqcxr16i54j43s259yan-python3.7-pyasn1-0.4.8/lib/python3.7/site-packages)
Found duplicated packages in closure for dependency 'packaging':
  packaging 20.3 (/nix/store/qvpsgs0sl3sh3knm5b1ya49x5zlf9v6j-python3-3.7.6/lib/python3.7/site-packages)
  packaging 20.1 (/nix/store/wrhaln323yi4i9d4s51ky92sj1r2cyky-python3.7-packaging-20.1/lib/python3.7/site-packages)
Found duplicated packages in closure for dependency 'idna':
  idna 2.9 (/nix/store/qvpsgs0sl3sh3knm5b1ya49x5zlf9v6j-python3-3.7.6/lib/python3.7/site-packages)
  idna 2.8 (/nix/store/g38lfy4g4k5g0aaipzrmql9nzrbl3cn7-python3.7-idna-2.8/lib/python3.7/site-packages)
Found duplicated packages in closure for dependency 'certifi':
  certifi 2019.11.28 (/nix/store/qvpsgs0sl3sh3knm5b1ya49x5zlf9v6j-python3-3.7.6/lib/python3.7/site-packages)
  certifi 2019.11.28 (/nix/store/gr6i7d5zbw6x62k7i0vzkfcdff3rqrbb-python3.7-certifi-2019.11.28/lib/python3.7/site-packages)
```

Reply:
Diff and build.log https://gist.github.com/mweinelt/5b12577a07e952157000247c536f37eb

Reply:
I'm getting the same result as @mweinelt. @WhittlesJr Do you have sandboxing disabled by any chance?